core/src/test/scala/org/apache/spark/util/AccumulatorV2Suite.scala

```diff
@@ -86,4 +86,48 @@ class AccumulatorV2Suite extends SparkFunSuite {
     assert(acc.sum == 5.0)
     assert(acc.avg == 1.25)
   }
+
+  test("ListAccumulator") {
+    val maxI = 1000
+    for (nThreads <- List(1, 10)) {
+      // test single & multi-threaded
+      sc = new SparkContext("local[" + nThreads + "]", "test")
+      val acc = sc.listAccumulator[Double]
+      assert(acc.isZero)
+
+      val d = sc.parallelize((1 to maxI) ++ (1 to maxI)).map(_.toDouble)
+      d.foreach { x => acc.add(x) }
+
+      assert(acc.value.size() === 2*maxI)
+      for (i <- 1 to maxI) {
+        assert(acc.value.contains(i.toDouble))
+      }
+      resetSparkContext()
+    }
+  }
+
+  test("LegacyAccumulatorWrapper") {
+    val acc = new LegacyAccumulatorWrapper("default", AccumulatorParam.StringAccumulatorParam)
+    assert(acc.value === "default")
+    assert(!acc.isZero)
+
+    acc.add("foo")
+    assert(acc.value === "foo")
+    assert(!acc.isZero)
+
+    // test using non-specialized add function
```

Comment:
string is not primitive so i don't think there is a difference between specialized / non-specialized. you can just remove this